include/bluetooth/services/otc.h

```diff
@@ -106,21 +123,25 @@ struct bt_otc_instance_t {
 	struct bt_otc_obj_metadata cur_object;
```

Comment:
It seems not ideal that there are two different types for metadata but I can't think of an easy way to combine the two at the moment. Although we could reuse the `struct bt_ots_obj_size` for the current and allocated size metadata fields.